### ISSUE TITLE
[v3.9.3+fabric-1.20.4] Non-English character endocing causing a crash in ItemStatsTab

### ISSUE THREAD

**Issue by Lopeww** [TheCSMods/mc-better-stats#113]
### What happened?

As I said in the titles I went to check stats and when I clicked for block stats my game crashed, I tried a second time and it crashed again. I leave the log here, maybe it conflicts with a mod

https://mclo.gs/KKt0OSk

### Steps to reproduce

1. Go to...
2. Then do...
3. See the bug or issue or a crash

### Relevant log output or crash report

```shell
https://mclo.gs/KKt0OSk
```

### Other installed mods

	   \-- com_electronwill_night-config_toml 3.6.6
	- litematica 0.17.0
	- malilib 0.18.1
	- memoryleakfix 1.1.5
	- minecraft 1.20.4
	- modernfix 5.14.0+mc1.20.4
	- modmenu 9.0.0
	- morechathistory 1.3.0
	- moreculling 1.20.4-0.22.1
	   |-- conditional-mixin 0.3.2
	   \-- mixinsquared 0.1.1
	- mousetweaks 2.25
	- nochatreports 1.20.4-v2.5.0
	- puzzleslib 20.4.40
	- replaymod 1.20.4-2.6.15
	- shulkerboxtooltip 4.0.8+1.20.4
	- sodium 0.5.8+mc1.20.4
	- spark 1.10.58
	   \-- fabric-permissions-api-v0 0.2-SNAPSHOT
	- starlight 1.1.3+fabric.f5dcd1a
	- visuality 0.7.3+1.20.4
	- visualworkbench 20.4.1
	- worldedit 7.3.0+6678-55745ad
	- yet_another_config_lib_v3 3.3.2+1.20.4
	   |-- com_twelvemonkeys_common_common-image 3.10.0
	   |-- com_twelvemonkeys_common_common-io 3.10.0
	   |-- com_twelvemonkeys_common_common-lang 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-core 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-metadata 3.10.0
	   |-- com_twelvemonkeys_imageio_imageio-webp 3.10.0
	   |-- org_quiltmc_parsers_gson 0.2.1
	   \-- org_quiltmc_parsers_json 0.2.1
	- zoomify 2.13.0
	   |-- com_akuleshov7_ktoml-core-jvm 0.5.1
	   |-- dev_isxander_settxi_settxi-core 2.10.6
	   \-- dev_isxander_settxi_settxi-kotlinx-serialization 2.10.6

### Code of Conduct

- [X] I agree that if the issue is regarding a crash, I gave the logs and the crash report.
- [X] I agree that I will be available later for any follow-up questions to help diagnose and resolve the issue.

**Comment by TheCSDev** [comment]
Hello and thanks a lot for the bug report!  
I have a quick question while I resolve this; Is your game or operating system language setting set to either "Turkish" or some other non-English language, or is your general locale setting "Turkish" or non-English?

As I looked into the bug report, the crash appears to be caused by a character encoding issue. For some strange and unexpected reason, the `I` character is being encoded as `\u0131` (aka `ı`), which appears to be a non-English lowercase `I` letter. This strange letter then ends up in a place that requires only English `A-z` characters, resulting in a crash (because `ı` is not English).

So tldr is, a string of text that says `ItemStatsTab` is converted to lowercase, and Java for some reason gives `ıtemstatstab` instead of `itemstats` tab, resulting in the crash (because `ı` is unexpected.

Anyways thanks again for the bug report, and have a great day! I will make sure to fix this so non-English users don't have to deal with this weird bug that wasn't even supposed to exist.

**Comment by Lopeww** [comment]
Just system lang Turkish. This happened to me in minecraft plugins before, I reached the plugin creator and he said he would fix it with localization in the same way and fixed it, if it helps... Thanks!

**Comment by TheCSDev** [comment]
Alright, thanks for confirming!  
Also I have published the patch, and everything should work fine now.
